pkg/tap/tap.go

```diff
@@ -37,7 +38,7 @@ func Reader(k8sAPI *k8s.KubernetesAPI, req *pb.TapByResourceRequest, timeout tim
 	if err != nil {
 		return nil, nil, err
 	}
-	url.Path = protohttp.TapReqToURL(req)
+	url.Path = path.Join(url.Path, protohttp.TapReqToURL(req))
```

Comment:
This looks good but I wonder if we're going to run into this again for any other requests we make directly to the Kubernetes API server in the future.  Do you think we could have a more general solution by adding a RoundTripper to the client in `k8sAPI.NewClient()` which prepends url path prefix to request paths?

Reply:
That already works correctly. This specific instance doesn’t work because we’re building the URL specific to tap requests and replacing the already constructed one by `NewClient()` wholesale here.

Reply:
ah, you're right!  thanks!

Reply:
I figured out the cause for the error, and it is not unique to rancher.

According to the `url.Join` [docs](https://golang.org/pkg/path/#Join):

```
Join joins any number of path elements into a single path, adding a separating slash if necessary. The result is Cleaned; in particular, all empty strings are ignored.
```

This is relevant because the path for the kubernetes API uses two slashes in a row when an item is "namespaced". 

Here are examples of the commands that are executed and the k8s API URLs that are generated when
`url.Path = path.Join(url.Path, protohttp.TapReqToURL(req))` is replaced with 
`url.Path = fmt.Sprintf("%s%s",url.Path, protohttp.TapReqToURL(req))`:

command:
`
bin/go-run cli tap -n linkerd deploy/linkerd-controller --verbose
`
URL:
`
https://<rancher-url>/k8s/clusters/c-66q2l/apis/tap.linkerd.io/v1alpha1/watch/namespaces/linkerd/deployments/linkerd-controller/tap
`

command:
`
bin/go-run cli tap -n linkerd deploy --verbose
`
URL:
`https://<rancher-url>/k8s/clusters/c-66q2l/apis/tap.linkerd.io/v1alpha1/watch/namespaces/linkerd/deployments//tap
`

Notice the consecutive slashes `//` between `deployments` and `tap`


I get the same results when running against a single node `kind` cluster on my local machine.

To summarize, the *gotcha* here is that `join.Path` calls `join.Clean` which results in consecutive slashes `//` being _cleaned_ to `/`